### PR TITLE
[RAIN-1687] PGR-migration script patch (tested locally).

### PR DESCRIPTION
**Pull request by shashwat-egov**

### CODE REVIEW

File: pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java [hunk 1]
```diff
@@ -189,22 +221,28 @@ private void setStatusToUUIDMap(){
         String serviceCode = serviceV1.getServiceCode();
         String serviceRequestId = serviceV1.getServiceRequestId();
         String description = serviceV1.getDescription();
-        String source = serviceV1.getSource().toString();
+        String source;
+        if(!ObjectUtils.isEmpty(serviceV1.getSource()))
+            source = serviceV1.getSource().toString();
+        else
+            source = "web";
```
Comment (talele08) [inline pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java:228]: Why default value is web?

File: pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java [hunk 2]
```diff
@@ -54,6 +55,21 @@
         }
     };
```
Comment (talele08) [inline pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java:57]: add the updated states which I have sent you

File: pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java [hunk 3]
```diff
@@ -216,14 +254,14 @@ private void setStatusToUUIDMap(){
          * Transform address and set geo location
          */
         GeoLocation geoLocation = GeoLocation.builder().longitude(longitutude).latitude(latitude).build();
-        org.egov.pgr.web.models.Address address = transformAddress(serviceV1.getAddressDetail());
-        address.setGeoLocation(geoLocation);
-        address.setTenantId(tenantId);
+        log.info("Address details: " + serviceV1.getAddressDetail());
+        org.egov.pgr.web.models.Address address = null;
+        if(serviceV1.getAddressDetail() != null) {
```
Comment (talele08) [inline pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java:259]: how many addressDetails are null?

File: pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java [hunk 4]
```diff
@@ -298,7 +331,7 @@ private ProcessInstance transformAction(ActionInfo actionInfo, Map<Long, String>
         String tenantId = actionInfo.getTenantId();
         Long createdTime = actionInfo.getWhen();
         String businessId = actionInfo.getBusinessKey();
-        String action = actionInfo.getAction();
+        String action = (actionInfo.getAction() != null) ? oldToNewAction.get(actionInfo.getAction()) : "COMMENT";
```
Comment (talele08) [inline pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java:334]: use StringUtils.isEmpty() to check for null

File: pgr-services/src/main/java/org/egov/pgr/util/MigrationUtils.java [hunk 5]
```diff
@@ -54,23 +57,25 @@ public MigrationUtils(UserUtils userUtils, PGRConfiguration config, ObjectMapper
          * @return
          */
 
+        ids.removeAll(Collections.singleton(null));
```
Comment (talele08) [inline pgr-services/src/main/java/org/egov/pgr/util/MigrationUtils.java:60]: why is this added?

**Review by talele08 — CHANGES_REQUESTED** [review]

File: pgr-services/src/main/java/org/egov/pgr/util/MigrationUtils.java (hunk 5 above)
Comment (shashwat-egov) [reply pgr-services/src/main/java/org/egov/pgr/util/MigrationUtils.java:60]: We are passing accountid attribute inside this List of ids data structure as well in which 128 records have the accountid as null. So, I have added this line to remove all the null values as no UUID corresponding to those exists.

File: pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java (hunk 3 above)
Comment (shashwat-egov) [reply pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java:259]: none of them are, actually earlier while testing some of the locality values were null, so I have handled that in the code itself so this "if" check is redundant, will remove it.

File: pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java (hunk 1 above)
Comment (shashwat-egov) [reply pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java:228]: When serviceV1.getSource() is null, should I let the source String have the value null?

File: pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java (hunk 2 above)
Comment (shashwat-egov) [reply pgr-services/src/main/java/org/egov/pgr/service/MigrationService.java:57]: have added the same one here

**Review by talele08 — APPROVED** [review]
